### ISSUE TITLE
Generation failed: Class 'JavaFile' has no instance getter 'path'.

### ISSUE THREAD

**Issue by NoNotNow** [dart-lang/dartdoc#1247]
After creating a new Project using darts "Uber Simple Web Application" sample content, run the dartdoc command in the root folder. The Generation fails with this output:

```
Generating documentation for 'untitled1' into C:\Users\mriveire\WebstormProjects\untitled1\doc\api\

Generation failed: Class 'JavaFile' has no instance getter 'path'.

NoSuchMethodError: method not found: 'path'
Receiver: Instance of 'JavaFile'
Arguments: []
dart:core                                                                                        Object.noSuchMethod
package:analyzer/source/pub_package_map_provider.dart 173                                        PubPackageMapProvider._runPubListDefault
package:analyzer/source/pub_package_map_provider.dart 74                                         PubPackageMapProvider.computePackageMap
package:dartdoc/dartdoc.dart 150                                                                 DartDoc._parseLibraries
package:dartdoc/dartdoc.dart 93                                                                  DartDoc.generateDocs.<async>
===== asynchronous gap ===========================
dart:async                                                                                       _Completer.completeError
package:dartdoc/dartdoc.dart 139                                                                 DartDoc.generateDocs.<async>
===== asynchronous gap ===========================
dart:async                                                                                       Future.Future.microtask
package:dartdoc/dartdoc.dart                                                                     DartDoc.generateDocs
E:\b\build\slave\dart-sdk-windows-stable\build\sdk\third_party\pkg\dartdoc\bin\dartdoc.dart 142  main.<async>.<fn>.<async>
===== asynchronous gap ===========================
package:stack_trace                                                                              Chain.capture
E:\b\build\slave\dart-sdk-windows-stable\build\sdk\third_party\pkg\dartdoc\bin\dartdoc.dart 141  main.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/dartdoc.dart 50                                                                  initGenerators.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/html_generator.dart 67                                                  HtmlGenerator.create.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/templates.dart 115                                                      Templates.create.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/templates.dart 98                                                       Templates.create.<async>._loadTemplate.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/resource_loader.dart 22                                                 loadAsString.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/resource_loader.dart 33                                                 loadAsBytes.<async>


C:\Users\mriveire\WebstormProjects\untitled1>dartdoc
Generating documentation for 'untitled1' into C:\Users\mriveire\WebstormProjects\untitled1\doc\api\

Generation failed: Class 'JavaFile' has no instance getter 'path'.

NoSuchMethodError: method not found: 'path'
Receiver: Instance of 'JavaFile'
Arguments: []
dart:core                                                                                        Object.noSuchMethod
package:analyzer/source/pub_package_map_provider.dart 173                                        PubPackageMapProvider._runPubListDefault
package:analyzer/source/pub_package_map_provider.dart 74                                         PubPackageMapProvider.computePackageMap
package:dartdoc/dartdoc.dart 150                                                                 DartDoc._parseLibraries
package:dartdoc/dartdoc.dart 93                                                                  DartDoc.generateDocs.<async>
===== asynchronous gap ===========================
dart:async                                                                                       _Completer.completeError
package:dartdoc/dartdoc.dart 139                                                                 DartDoc.generateDocs.<async>
===== asynchronous gap ===========================
dart:async                                                                                       Future.Future.microtask
package:dartdoc/dartdoc.dart                                                                     DartDoc.generateDocs
E:\b\build\slave\dart-sdk-windows-stable\build\sdk\third_party\pkg\dartdoc\bin\dartdoc.dart 142  main.<async>.<fn>.<async>
===== asynchronous gap ===========================
package:stack_trace                                                                              Chain.capture
E:\b\build\slave\dart-sdk-windows-stable\build\sdk\third_party\pkg\dartdoc\bin\dartdoc.dart 141  main.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/dartdoc.dart 50                                                                  initGenerators.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/html_generator.dart 67                                                  HtmlGenerator.create.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/templates.dart 115                                                      Templates.create.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/templates.dart 98                                                       Templates.create.<async>._loadTemplate.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/resource_loader.dart 22                                                 loadAsString.<async>
dart:async                                                                                       _SyncCompleter.complete
package:dartdoc/src/html/resource_loader.dart 33                                                 loadAsBytes.<async>
```

**Comment by devoncarew** [comment]
Are you running the version of dartdoc out of the SDK? What version of the SDK?

**Comment by NoNotNow** [comment]
I'm running it from the command line on Windows 8.1
`$C:\_dev\webclient\webclient>dartdoc`
My dart SDK version is 1.19.0

**Comment by hoylen** [comment]
FYI, I'm getting the same problem on OS X 10.11.6.

_Generation failed: Class 'JavaFile' has no instance getter 'path'_

- dartdoc version: 0.9.6+2
- Dart VM version: 1.19.0 (Fri Aug 26 02:27:39 2016) on "macos_x64"

The docs generate fine using the older dartdoc 0.9.0 with Dart VM version 1.17.1.

**Comment by devoncarew** [comment]
Thanks for the reports! We'll take a look at this.

**Comment by devoncarew** [comment]
This was fixed in #1234; we need to patch https://github.com/dart-lang/sdk/issues/27184 into the stable sdk to get this fix to users.
